modules/login/src/main/resources/META-INF/resources/login/js/login.js

```diff
@@ -0,0 +1,64 @@
+var loginpage = function($, w, moment) {
+
+    var pageData = {};
+
+    function setPageDate(data) {
+        pageData = data;
+    }
+    function switchWorkspace(selectobj) {
+        // document.getElementById("currentWorkspace").value = newWorkspace;
+        if ($(selectobj.target).val() != " ") {
+            $('#currentWorkspace').val($(selectobj.target).val());
+            document.forms["frmWorkspace"].submit();
+        }
+    }
+
+    function setWorkspace() {
+        //set the list of workspaces.
+        pageData.workspaces.forEach(function (item) {
+            $('#server').append('<option value="' + item.id + '">' + item.name + '</option>')
+        });
+        if (pageData.workspaces.length < 2) { $('#workspaces').addClass('hidden');} //Don't show if there is only one workspace.
```

Comment:
Same idea here. Extracting the conditional logic to a well-named function can help readability.